bot/cogs/coding/reference.py

```diff
@@ -108,14 +113,15 @@ async def _git_main_ref(part: str, ctx: Context, text: str) -> Message:
                 content = "\n".join([markdownify(p) for p in tag.find_all(lambda x: x.name in ["p", "pre"])])
                 embed.add_field(name=tag.find("h2").text, value=content[:1024], inline=False)
 
+            await client_session.close()
             return await ctx.send(embed=embed)
 
 
 git_ref = partial(_git_main_ref, "git-")
 git_tutorial_ref = partial(_git_main_ref, "")
 
 
-async def sql_ref(ctx: Context, text: str) -> Message:
+async def sql_ref(bot: Bot, ctx: Context, text: str) -> Message:
```

Comment:
why r u adding bot as a Param everywhere?

Reply:
Its pretty easy to find out why by looking at the code.
But the reason is because the aiohttp session is inside of the bot.
The reason why we do this to ensure that the session is properly closed in the shutoff command.